app/lib/activitypub/activity/like.rb

```diff
@@ -3,12 +3,38 @@
 class ActivityPub::Activity::Like < ActivityPub::Activity
   def perform
     original_status = status_from_uri(object_uri)
+    return if original_status.nil? || !original_status.account.local? || delete_arrived_first?(@json['id'])
 
-    return if original_status.nil? || !original_status.account.local? || delete_arrived_first?(@json['id']) || @account.favourited?(original_status)
+    return if maybe_process_misskey_reaction
+
+    return if @account.favourited?(original_status)
 
     favourite = original_status.favourites.create!(account: @account)
 
     LocalNotificationWorker.perform_async(original_status.account_id, favourite.id, 'Favourite', 'favourite')
     Trends.statuses.register(original_status)
   end
+
+  # Misskey delivers reactions as likes with the emoji in _misskey_reaction
+  # see https://misskey-hub.net/ns.html#misskey-reaction for details
```

Comment:
This only checks the `_misskey_reaction` attribute (defined by Misskey), even though the `content` attribute (defined by ActivityPub) has been provided by Misskey [for a long time (since 2020-02-06)](https://github.com/misskey-dev/misskey/commit/988ac80087a116173ab36043a7a9dd2228d2986c#diff-12e8576204846b63bad4472ff3737065ab6985294a984a51bb320628161c2cd2) and [is still provided](https://github.com/misskey-dev/misskey/blob/75034d9240c069baff5a24409ea172374261ea3b/packages/backend/src/core/activitypub/ApRendererService.ts#L274). Not checking the `content` field of a `Like` activity will make reactions from Foundkey not show up correctly.

Since the `content` attribute is defined by ActivityPub it should IMHO be preferred.

( originally posted in https://akkoma.dev/AkkomaGang/akkoma/issues/499 see also https://akkoma.dev/FoundKeyGang/FoundKey/issues/354 )

Reply:
Should be fixed with 265eeed.